diff --git a/configure.in b/configure.in
index f12af2a..01fca32 100644
--- a/configure.in
+++ b/configure.in
@@ -15,6 +15,8 @@ AC_PROG_CC
 AM_PROG_LIBTOOL
 AC_PROG_MAKE_SET
 
+PKG_PROG_PKG_CONFIG
+
 dnl intitialization
 if test "x$prefix" = "xNONE"; then
     prefix="/usr/local"
@@ -43,6 +45,18 @@ dnl ugly ugly hack
 AC_CHECK_LIB(c, err, [ ERRO="" ], [ ERRO="err.o" ],)
 AC_SUBST(ERRO)
 
+AC_MSG_CHECKING([for libtirpc])
+if "$PKG_CONFIG" libtirpc ; then
+  LIBTIRPC_CFLAGS="$("$PKG_CONFIG" --cflags libtirpc)"
+  CFLAGS="$LIBTIRPC_CFLAGS $CFLAGS"
+  LIBTIRPC_LIBS="$("$PKG_CONFIG" --libs libtirpc)"
+  LIBS="$LIBTIRPC_LIBS $LIBS"
+  AC_MSG_RESULT([\$CFLAGS: $LIBTIRPC_CFLAGS \$LIBS: $LIBTIRPC_LIBS])
+else
+  AC_MSG_RESULT([not found])
+  AC_MSG_ERROR([libtirpc not found, but is required])
+fi
+
 dnl Checks for header files.
 AC_HEADER_STDC
 AC_HEADER_SYS_WAIT
