commit 2a74dbe99cf085729facbe3baad02d77e338777c
Author: inty <interruptinuse@gmail.com>
Date:   Sun Apr 21 16:48:52 2019 +0300

    dedupe

diff --git a/imguralbum.py b/imguralbum.py
index 2137a8d..2e1fbf8 100755
--- a/imguralbum.py
+++ b/imguralbum.py
@@ -42,6 +42,16 @@ class ImgurAlbumException(Exception):
 
 
 class ImgurAlbumDownloader:
+    def deduplicate(self):
+        dedup = []
+
+        for f in self.imageIDs:
+            if f not in dedup:
+                dedup += [f]
+
+        self.imageIDs = dedup
+
+
     def __init__(self, album_url):
         """
         Constructor. Pass in the album_url that you want to download.
@@ -77,6 +87,8 @@ class ImgurAlbumDownloader:
         html = self.response.read().decode('utf-8')
         self.imageIDs = re.findall('.*?{"hash":"([a-zA-Z0-9]+)".*?"ext":"(\.[a-zA-Z0-9]+)".*?', html)
 
+        self.deduplicate()
+
         self.cnt = Counter()
         for i in self.imageIDs:
             self.cnt[i[1]] += 1
