--- a/droopy
+++ b/droopy
@@ -78,7 +78,7 @@ else:
 import cgi
 import os
 import posixpath
-import macpath
+import os.path
 import ntpath
 import argparse
 import mimetypes
@@ -109,7 +109,7 @@ def fullpath(path):
 
 def basename(path):
     "Extract the file base name (some browsers send the full file path)."
-    for mod in posixpath, macpath, ntpath:
+    for mod in posixpath, os.path, ntpath:
         path = mod.basename(path)
     return path
 
